tap_zendesk/client.py

```diff
@@ -129,8 +121,135 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
         Yields:
             Each record from the source.
         """
-        # TODO: Parse response body and return a set of records.
-        yield from extract_jsonpath(self.records_jsonpath, input=response.json())
+        if not response.content:
+            self.logger.warning("Received empty response for URL: %s", response.url)
+            return
+
+        try:
+            response_json = response.json()
+            self.logger.debug(f"Response JSON: {response_json}")
+        except json.JSONDecodeError:
+            self.logger.error("Unable to decode JSON response: %s", response.text)
+            return
+
+        yield from extract_jsonpath(self.records_jsonpath, input=response_json)
+
+    def _request(
+        self,
+        prepared_request: PreparedRequest,
+        context: Optional[Dict[str, Any]] = None,
+    ) -> Response:
+        """Send a HTTP request with rate limiting.
+
+        Args:
+            prepared_request: The prepared HTTP request.
+            context: Stream partition or context dictionary.
+
+        Returns:
+            The HTTP response object.
+        """
+        # Calculate the time difference between the current time and the last request time
+        time_since_last_request = time.time() - self.last_request_time
+
+        # Ensure we don't exceed 250 requests per minute (1 request per 0.24 seconds)
+        if time_since_last_request < 0.24:
+            time.sleep(0.24 - time_since_last_request)
```

Comment:
I think we don't need to force the number of requests per minute, because in case the number of requests limit grows or shrinks, if we control only the min number of requests we will have no issue with it.

Reply:
> if we control only the min number of requests we will have no issue with it.

OK, but could you clarify more? How would you implement this?

Reply:
There is a remain rate limit in the response header, so we can always check if it below the minimum value:

https://github.com/Automattic/tap-zendesk/blob/0c17e64dbd453ad0c0e9e6c9330bff8c3875619c/tap_zendesk/__init__.py#L59-L68

Reply:
Thank you! I implemented this function in [e1263b4](https://github.com/Automattic/tap-zendesk-singer-sdk/pull/1/commits/e1263b464ef56455ce13b0ff8feb7d827cce832f).